test/parallel_testsuite.py

```diff
@@ -182,38 +193,40 @@ def stopTest(self, test):
     self.buffered_result.duration = self.test_duration
 
   def compute_progress(self):
+    if not self.lock:
+      return ''
     with self.lock:
-      val = f'[{int(self.progress_counter.value * 100 / self.num_tests)}%]'
+      val = f'[{int(self.progress_counter.value * 100 / self.num_tests)}%] '
       self.progress_counter.value += 1
     return val
 
   def addSuccess(self, test):
-    print(self.compute_progress(), test, '... ok (%.2fs)' % (self.calculateElapsed()), file=sys.stderr)
+    print(f'{self.compute_progress()}{test}', '... ok (%.2fs)' % (self.calculateElapsed()), file=sys.stderr)
```

Comment:
Maybe just use a single f-string here to avoid mixing the two styles on the same line:

```
f'{self.compute_progress()}{test} ... ok ({self.calculateElapsed():.2f})'
```